src/Nancy/AfterPipeline.cs

```diff
@@ -83,6 +83,12 @@ private static void ExecuteTasksInternal(NancyContext context, CancellationToken
             {
                 var current = enumerator.Current.Invoke(context, cancellationToken);
 
+                if (current == null)
+                {
+                    var method = enumerator.Current.Method ?? (object)enumerator.Current;
+                    throw new InvalidOperationException(string.Format("'{0}' returned null; a non-null {1} expected.", method, typeof(Task)));

```

Comment:
Found it in the CI build log, and as I thought - not very nice :cry: 

> 'System.Threading.Tasks.Task <Pipeline_containing_method_returning_null_from_lambda_throws_InvalidOperationException>b__1a(Nancy.NancyContext, System.Threading.CancellationToken)' returned null; a non-null System.Threading.Tasks.Task expected.

I'd rather remove the parameters and give a hard coded message.